### ISSUE TITLE
RFC: Read REPO_BRANCH_CONTEXT from a json file

### ISSUE THREAD

**Issue by atodorov** [cockpit-project/cockpit#11934]
Now that we have other repositories using `bots/` and depending on Cockpit CI infra, namely lorax, I think it makes sense to try and read the context from a file inside the current repository, not hard-code it in `bots/tests-scan`.

Shouldn't be too hard to implement, let me know what you think ?

**Comment by martinpitt** [comment]
Changing a json file in the cockpit repo isn't significantly easier than changing the map in tests-scan. What we really want is to move that map out into the projects, so that it doesn't have to be maintained centrally, and a change in tests and the corresponding test map can happen in lockstep. We didn't design this yet, but it's on our medium-term roadmap.

**Comment by atodorov** [comment]
FTR. I meant a json file inside each and every repo, not inside the cockpit repository.

**Comment by martinpitt** [comment]
@atodorov : Yes, something like this, like ~/.cockpit-tests.json. We need to do some ground work for that to happen, like webhook not calling tests-scan by itself any more, but deferring this to the tasks bots.
